test/e2e/ic_conditions_metric_test.go

```diff
@@ -0,0 +1,179 @@
+func waitForIngressControllerConditionsMetrics(t *testing.T, prometheusClient prometheusv1.API) error {
+	t.Logf("Waiting for ingress_controller_conditions to be present")
+	if err := wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 2*time.Minute, false, func(context context.Context) (bool, error) {
+		result, _, err := prometheusClient.Query(context, "ingress_controller_conditions", time.Now())
+		if err != nil {
+			t.Logf("Failed to fetch metrics: %v, retrying...", err)
+			return false, nil
+		}
+
+		// Check if fetched metrics is of Vector type.
+		vector, ok := result.(model.Vector)
+		if !ok {
+			t.Logf("Unexpected metric type, retrying...")
+			return false, nil
+		}
+
+		// Check if length of returned metric Vector is zero.
+		if len(vector) == 0 {
+			t.Logf("Metric is empty, retrying...")
+			return false, nil
+		}
+
+		return true, nil
+	}); err != nil {
+		return fmt.Errorf("Error waiting for ingress controller metrics: %w", err)
+	}
+	return nil
+}
+
+// restartOperatorPod will restart the operator pod, and if some error occurs it returns it
+func restartOperatorPod(t *testing.T, kubeClient kubernetes.Interface) error {
+	interval, timeout := 5*time.Second, 5*time.Minute
+	var podsList *corev1.PodList
+
+	// Find the operator pod
+	t.Logf("Restarting Ingress operator pod...")
+	if err := wait.PollUntilContextTimeout(context.Background(), interval, timeout, false, func(context context.Context) (bool, error) {
+		innerPodsList, err := kubeClient.CoreV1().Pods("openshift-ingress-operator").List(context, metav1.ListOptions{})
+		podsList = innerPodsList
+		if err != nil {
+			t.Logf("Failed to list pods: %v, retying...", err)
+			return false, err
+		}
+		return innerPodsList != nil && len(innerPodsList.Items) > 0, nil
+	}); err != nil {
+		return err
+	}
+	operatorPodName, operatorPodUID := extractIngressOperatorPodNameAndUID(podsList)
+
+	if operatorPodName == "" || operatorPodUID == "" {
+		return errors.New(("Unable to find ingress operator pod"))
+	}
+
+	// Delete the operator pod
+	if err := wait.PollUntilContextTimeout(context.Background(), interval, timeout, false, func(context context.Context) (bool, error) {
+		if err := kubeClient.CoreV1().Pods("openshift-ingress-operator").Delete(context, operatorPodName, metav1.DeleteOptions{}); err != nil {
+			t.Logf("Failed to delete operator pod: %v, retying...", err)
+			return false, err
+		}
+		return true, nil
+	}); err != nil {
+		return err
+	}
+
+	// Wait for new pod to be ready
+	t.Logf("Polling for up to %v to verify that the operator restart is terminated...", timeout)
+	if err := wait.PollUntilContextTimeout(context.Background(), interval, timeout, false, func(context context.Context) (bool, error) {
+		podsList, err := kubeClient.CoreV1().Pods("openshift-ingress-operator").List(context, metav1.ListOptions{})
+		if err != nil || len(podsList.Items) == 0 {
+			return false, err
+		}
```

Comment:
```suggestion
		if err != nil {
		    t.Logf("Failed to list pods: %v, retying...", err) 
			return false, nil
		}
		
		 if podsList == nil || len(podsList.Items) == 0 {
		    t.Logf("No pods found, retying...") 
			return false, nil
		 }
```
1. Returning `nil` instead of the error, for the same reason as the previous comment.
2. Aligning the logic about `podsList` with the previous list polling.